### PR TITLE
No allocations on `EncodeTime`, run benchmarks

### PR DESCRIPTION
**Pull request by smira**
Use per-instance lazily-allocated buffer to avoid allocations
in `EncodeTime`. We can't use use `e.buf`, as it's used to
encode extLen, and it's smaller than required (9 bytes).
Using buffer on stack doesn't work, as Go doesn't have
a way to figure out if `e.write(b)` escapes or not, so it thinks
it always escapes.

Run benchmarks for `make` (and in Travis CI), and fix benchmark
which was failing probably due to some changes in the codebase.

Benchmark, before:

```
BenchmarkTime-12    	20000000	        94.0 ns/op	       8 B/op	       1 allocs/op
```

after:

```
BenchmarkTime-12    	20000000	        76.7 ns/op	       0 B/op	       0 allocs/op
```

### CODE REVIEW

File: encode.go [hunk 1]
```diff
@@ -53,6 +53,8 @@ type Encoder struct {
 	w   writer
 	buf []byte
 
+	tbuf []byte
```
Comment (vmihailenco) [inline encode.go]: Please group this with other `buf` var and rename to `timeBuf`. Also having a comment why we need another buffer ("We can't use use e.buf, as it's used to encode extLen") will help future readers a lot.

Comment (smira) [reply encode.go]: thanks, updated

**Comment by vmihailenco** [comment]
Thank you!

**Comment by smira** [comment]
would you mind tagging new release with this commit please?

**Comment by vmihailenco** [comment]
Tagged v4.0.4

**Pull request merged**